packages/api/src/providers/__tests__/oauth1-deregister.test.ts

```diff
@@ -70,68 +41,164 @@ describe("oauth1-deregister", () => {
-    getUserTokenByUATMock.mockReturnValueOnce([userTokenMocked, anotherUserTokenMocked]);
-    saveUserTokenMock.mockReturnValueOnce(updatedUserToken);
-    updateProviderDataMock.mockReturnValueOnce(testUserModified);
+    getUserTokenByUATMock.mockReturnValueOnce([userToken_1, userToken_2]);
+    saveUserTokenMock.mockReturnValueOnce(userToken_1_upd);
+    updateProviderDataMock.mockReturnValueOnce(testUser_1_upd);
 
-    await testGarmin.deregister([testUser.providerMap.garmin.token], testUser.cxId);
+    await testGarmin.deregister([testUser_1.providerMap.garmin.token], testUser_1.cxId);
 
     expect(getUserTokenByUATMock).toHaveBeenCalledTimes(1);
     expect(getUserTokenByUATMock).toHaveBeenCalledWith({
-      oauthUserAccessToken: testUser.providerMap.garmin.token,
+      oauthUserAccessToken: testUser_1.providerMap.garmin.token,
     });
     expect(saveUserTokenMock).toHaveBeenCalledTimes(1);
+
+    const firstToken = saveUserTokenMock.mock.calls[0][0];
+
+    expect(firstToken).toEqual(
+      expect.objectContaining({
+        oauthUserAccessToken: undefined,
+        oauthUserAccessSecret: undefined,
+      })
+    );
+
     expect(updateProviderDataMock).toHaveBeenCalledTimes(1);
     expect(updateProviderDataMock).toHaveBeenCalledWith({
-      id: updatedUserToken.userId,
-      cxId: updatedUserToken.cxId,
+      id: userToken_1_upd.userId,
+      cxId: userToken_1_upd.cxId,
       provider: "garmin",
       providerItem: undefined,
     });
   });
 
   it("processes multiple tokens", async () => {
-    getUserTokenByUATMock.mockReturnValue([userTokenMocked]);
-    saveUserTokenMock.mockReturnValue(userTokenMocked);
-    updateProviderDataMock.mockReturnValue(testUserModified);
+    getUserTokenByUATMock.mockReturnValueOnce([userToken_1]);
+    getUserTokenByUATMock.mockReturnValueOnce([userToken_2]);
+    saveUserTokenMock.mockReturnValueOnce(userToken_1_upd);
+    saveUserTokenMock.mockReturnValueOnce(userToken_2_upd);
+    updateProviderDataMock.mockReturnValueOnce(testUser_1_upd);
+    updateProviderDataMock.mockReturnValueOnce(testUser_2_upd);
+
+    await testGarmin.deregister([
+      testUser_1.providerMap.garmin.token,
+      testUser_2.providerMap.garmin.token,
+    ]);
+
+    expect(getUserTokenByUATMock).toHaveBeenCalledTimes(2);
+    expect(getUserTokenByUATMock).toHaveBeenCalledWith({
+      oauthUserAccessToken: testUser_1.providerMap.garmin.token,
+    });
+    expect(getUserTokenByUATMock).toHaveBeenCalledWith({
+      oauthUserAccessToken: testUser_2.providerMap.garmin.token,
+    });
+
+    const firstTokenList = getUserTokenByUATMock.mock.calls[0][0];
+    const secondTokenList = getUserTokenByUATMock.mock.calls[1][0];
 
-    await testGarmin.deregister(
-      [testUser.providerMap.garmin.token, thirdTestUser.providerMap.garmin.token],
-      testUser.cxId
+    expect(firstTokenList).toEqual(
+      expect.objectContaining({
+        oauthUserAccessToken: "some_uat_string",
```

Comment:
Referring to the comment about trying not to separate the input/output from the tests, this literal doesn't tell me anything, I need to search through the code to understand this was passed as parameter to `getUserTokenByUATMock` and its coming from another file.

To minimize this, we could instead of having the literal here have something like this:

```ts
oauthUserAccessToken: userToken_1.oauthUserAccessToken,
```

Then I can easily refer to the var/const being passed to `getUserTokenByUATMock` just by clicking on it.